src/document.js

```diff
@@ -80,6 +80,46 @@ const hideLoadingIndicator = () => {
 	document.getElementById('proxyLoadingMessage').textContent = ''
 }
 
+const generateCSSVarTokens = () => {
+	/* NC versus COOL */
+	var cssVarMap = {
+		'--color-main-text': '--co-color-main-text',
+		'--color-main-background': '--co-body-bg',
+		'--color-primary-text': '--co-primary-text',
+		'--color-primary-element': '--co-primary-element:--co-text-accent',
+		'--color-primary-element-light': '--co-primary-element-light',
+		'--color-error': '--co-color-error',
+		'--color-warning': '--co-color-warning',
+		'--color-success': '--co-color-success',
+		'--border-radius': '--co-border-radius',
+		'--border-radius-large': '--co-border-radius-large',
+		'--color-background-hover': '--co-background-hover',
+		'--color-background-dark': '--co-background-dark',
+		'--color-text-light': '--co-text-light',
+		'--color-text-lighter': '--co-text-lighter',
+		'--color-loading-light': '--co-loading-light',
+		'--color-loading-dark': '--co-loading-dark',
+		'--color-box-shadow': '--co-box-shadow',
+		'--color-border': '--co-border',
+		'--color-border-dark': '--co-border-dark',
+		'--border-radius-pill': '--co-border-radius-pill'
+	}
+	var str = ''
+	for (var cssVarKey in cssVarMap) {
+		var cStyle = window.parent.getComputedStyle(document.documentElement).getPropertyValue(cssVarKey)
+		if (!cStyle) {
+			// try suffix -dark instead
+			cStyle = window.parent.getComputedStyle(document.documentElement).getPropertyValue(cssVarKey + '-dark')
```

Comment:
```suggestion
			cStyle = window.getComputedStyle(document.documentElement).getPropertyValue(cssVarKey + '-dark')
```

Reply:
Thank you for the reply. Initially it was like this but later we discovered that it does not work on Firefox like that. The CSS variables become <empty string>. The only solution I found was using window.parent.

Reply:
Indeed seems to be a firefox bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1471231

Then let's get this in as it is and maybe see if we can work around that by not using `display: none;` but opacity of 0 until the frame is loaded.